vms/avm/tx_builders.go

```diff
@@ -0,0 +1,229 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package avm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/set"
+	"github.com/ava-labs/avalanchego/vms/avm/txs"
+	"github.com/ava-labs/avalanchego/vms/components/avax"
+	"github.com/ava-labs/avalanchego/vms/components/verify"
+	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
+	"github.com/ava-labs/avalanchego/wallet/chain/x/signer"
+	"github.com/ava-labs/avalanchego/wallet/subnet/primary/common"
+
+	walletbuilder "github.com/ava-labs/avalanchego/wallet/chain/x/builder"
+)
+
+type txBuilderBackend interface {
+	walletbuilder.Backend
+	signer.Backend
+
+	Context() *walletbuilder.Context
+	ResetAddresses(addrs set.Set[ids.ShortID])
+}
+
+func buildCreateAssetTx(
+	backend txBuilderBackend,
+	name, symbol string,
+	denomination byte,
+	initialStates map[uint32][]verify.State,
+	kc *secp256k1fx.Keychain,
+	changeAddr ids.ShortID,
+) (*txs.Tx, ids.ShortID, error) {
```

Comment:
I feel like returning `ids.ShortID` is strange because the caller already knows the `changeAddr`

Reply:
Agree, dropped